vars/perfRegressionBareMetalSinglePipeline.groovy

```diff
@@ -0,0 +1,329 @@
+                   description: 'if true, enable operator tls, and install haproxy ingress controller',
+                   name: 'k8s_enable_tls')
+        }
+        options {
+            timestamps()
+            disableConcurrentBuilds()
+            buildDiscarder(logRotator(numToKeepStr: '20'))
+        }
+        stages {
+            stage("Preparation") {
+                // NOTE: this stage is a workaround for the following Jenkins bug:
+                // https://issues.jenkins-ci.org/browse/JENKINS-41929
+                when { expression { env.BUILD_NUMBER == '1' } }
+                steps {
+                    script {
+                        if (currentBuild.getBuildCauses('hudson.model.Cause$UserIdCause') != null) {
+                            currentBuild.description = ('Aborted build#1 not having parameters loaded. \n'
+                              + 'Build#2 is ready to run')
+                            currentBuild.result = 'ABORTED'
+
+                            error('Abort build#1 which only loads params')
+                        }
+                    }
+                }
+            }
+            stage('Get test duration') {
+                steps {
+                    catchError(stageResult: 'FAILURE') {
+                        timeout(time: 10, unit: 'MINUTES') {
+                            script {
+                                wrap([$class: 'BuildUser']) {
+                                    dir('scylla-cluster-tests') {
+                                        checkout scm
+                                        (testDuration, testRunTimeout, runnerTimeout, collectLogsTimeout, resourceCleanupTimeout) = getJobTimeouts(params, builder.region)
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            stage("Checkout") {
+                steps {
+                    catchError(stageResult: 'FAILURE') {
+                        timeout(time: 5, unit: 'MINUTES') {
+                            script {
+                                wrap([$class: 'BuildUser']) {
+                                    dir('scylla-cluster-tests') {
+                                        checkout scm
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            stage("Create SCT Runner") {
+                steps {
+                    wrap([$class: 'BuildUser']) {
+                        dir('scylla-cluster-tests') {
+                            timeout(time: 10, unit: 'MINUTES') {
+                                script {
+                                    // Always use aws for sct runner
+                                    def awsParams = [:]
+                                    awsParams["backend"] = "aws"
```

Comment:
All the pipeline duplicated just for this configuration of the runner ?

I would suggest introducing an optional runner_backend option to SCT for that

Reply:
Yes, that would be a good thing to implement, should I do it with this PR or move it to a separate task?

Reply:
It depends if it's a 20-30min work, let's do it now, if you estimate it to be more then that, let's write a separate task for it

Reply:
Reverted that, changed to adjusting `getCloudProviderFromBackend` to map `baremetal` to `aws`

Reply:
Seems to be working fine now, agent is properly selected and sct runner is created using old params